File: internal/router/apiv1/comment.go
package apiv1

import (
	"github.com/cloudwego/hertz/pkg/route"
	"github.com/zyj/my-blog/internal/controller"
	"github.com/zyj/my-blog/internal/middleware"
	"github.com/zyj/my-blog/pkg/constant"
)

func registerCommentRoutes(group *route.RouterGroup) {
	comment := group.Group("/comment")
	comment.GET("/list", middleware.UseAuth(false), controller.ListComments)
	comment.GET("/:id/replies", middleware.UseAuth(false), controller.ListCommentReplies)
	comment.GET("/:id/moderation", middleware.UseAuth(true), controller.GetCommentModeration)
	comment.POST("", middleware.UseAuth(true), controller.CreateComment)
	comment.DELETE("/:id", middleware.UseAuth(true), controller.DeleteComment)

	admin := group.Group(
		"/admin/comment",
		middleware.UseAuth(true),
		middleware.UseRole(constant.RoleAdmin),
	)
	admin.GET("/list", controller.ListAdminComments)
	admin.PATCH("/:id/moderation", controller.ModerateComment)
	admin.DELETE("/:id", controller.DeleteAdminComment)
}

File: web/src/api/comment.ts
import type {
  AdminCommentListRequest,
  Comment,
  CommentListRequest,
  CommentListResponse,
  CommentModerationResponse,
  CreateCommentRequest,
} from '@/models/comment'
import type { Resp } from '@/models/resp'
import { axiosClient } from '@/lib/api/client'
import { unwrapResponse } from '@/lib/api/response'

export async function listComments(
  req: CommentListRequest,
): Promise<CommentListResponse> {
  const response = await axiosClient.get<Resp<CommentListResponse>>(
    '/comment/list',
    { params: req },
  )

  return unwrapResponse(response.data)
}

export async function listCommentReplies(id: number): Promise<Comment[]> {
  const response = await axiosClient.get<Resp<Comment[]>>(
    `/comment/${id}/replies`,
  )

  return unwrapResponse(response.data)
}

export async function createComment(
  req: CreateCommentRequest,
): Promise<Comment> {
  const response = await axiosClient.post<Resp<Comment>>(
    '/comment',
    req,
  )

  return unwrapResponse(response.data)
}

export async function getCommentModeration(
  id: number,
): Promise<CommentModerationResponse> {
  const response = await axiosClient.get<Resp<CommentModerationResponse>>(
    `/comment/${id}/moderation`,
  )
  return unwrapResponse(response.data)
}

export async function deleteComment(id: number): Promise<void> {
  await axiosClient.delete(`/comment/${id}`)
}

export async function listAdminComments(
  req: AdminCommentListRequest = {},
): Promise<CommentListResponse> {
  const response = await axiosClient.get<Resp<CommentListResponse>>(
    '/admin/comment/list',
    { params: req },
  )

  return unwrapResponse(response.data)
}

export async function deleteAdminComment(id: number): Promise<void> {
  await axiosClient.delete(`/admin/comment/${id}`)
}
